salt/minion.py

```diff
@@ -3777,3 +3777,154 @@ def _target(cls, minion_instance, opts, data, connected):
                 Minion._thread_multi_return(minion_instance, opts, data)
             else:
                 Minion._thread_return(minion_instance, opts, data)
+
+
+class SProxyMinion(SMinion):
+    '''
+    Create an object that has loaded all of the minion module functions,
+    grains, modules, returners etc.  The SMinion allows developers to
+    generate all of the salt minion functions and present them with these
+    functions for general use.
+    '''
+    def _load_modules(self, force_refresh=False, notify=False, grains=None):
+        '''
+        Return the functions and the returners loaded up from the loader
+        module
+        '''
+        # if this is a *nix system AND modules_max_memory is set, lets enforce
+        # a memory limit on module imports
+        # this feature ONLY works on *nix like OSs (resource module doesn't work on windows)
+        modules_max_memory = False
+        if self.opts.get('modules_max_memory', -1) > 0 and HAS_PSUTIL and HAS_RESOURCE:
+            log.debug(
+                'modules_max_memory set, enforcing a maximum of %s',
+                self.opts['modules_max_memory']
+            )
+            modules_max_memory = True
+            old_mem_limit = resource.getrlimit(resource.RLIMIT_AS)
+            rss, vms = psutil.Process(os.getpid()).memory_info()[:2]
+            mem_limit = rss + vms + self.opts['modules_max_memory']
+            resource.setrlimit(resource.RLIMIT_AS, (mem_limit, mem_limit))
+        elif self.opts.get('modules_max_memory', -1) > 0:
+            if not HAS_PSUTIL:
+                log.error('Unable to enforce modules_max_memory because psutil is missing')
+            if not HAS_RESOURCE:
+                log.error('Unable to enforce modules_max_memory because resource is missing')
```

Comment:
Can we rephrase these messages to be user-compatible? ".....psutil _Python package dependency_ is not installed". Same to "resource".

Reply:
Yeah, this isn't going to survive the cleanup, because it is a one off process, and this stuff is not needed.